roomserver/state/state.go

```diff
@@ -19,63 +19,1068 @@ package state
 import (
 	"context"
 	"errors"
+	"fmt"
+	"sort"
+	"time"
 
 	"github.com/matrix-org/dendrite/roomserver/state/database"
-	v1 "github.com/matrix-org/dendrite/roomserver/state/v1"
+	"github.com/matrix-org/util"
+	"github.com/prometheus/client_golang/prometheus"
 
 	"github.com/matrix-org/dendrite/roomserver/types"
 	"github.com/matrix-org/gomatrixserverlib"
 )
 
-type StateResolutionVersion int
+type StateResolution struct {
+	db database.RoomStateDatabase
+}
+
+func Prepare(db database.RoomStateDatabase) StateResolution {
+	return StateResolution{
+		db: db,
```

Comment:
It's not "preparing" anything (I intuitively expected to see SQL prepares here), please call it what it is, `NewStateResolution` or whatever.

Reply:
Done